proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,376 @@
+feature other than on the PR. It has no comments from AWS employees, other than
+to indicate that it is a “large” pull request, since 2017. Conversation
+continued for a year on the PR by interested users with no other interaction or
+guidance from AWS.
+
+A idea that has not been vetted by the community and the maintainers is
+difficult to approve; it's hard to have a conversation when it involves an idea
+proposal as well as code changes. It usually ends in an unmerged pull request or
+significant code revision, which is effort that was unnecessarily expended by
+the user and potentially the maintainer. We need a structured process for the
+life cycle of an idea that can culminate in a community contribution. We also
+need to prioritize issues against other requests from the community given the
+effort to contribute for users and review for maintainers.
+
+## Goals
+
+The approach for customer contributions should satisfy the following goals:
+
+1. Users should know how they can help. It should be straightforward for
+   community members to make high quality contributions.
+1. No requests should exist without a response. Communication from maintainers
+   should happen in a timely manner, even (especially!) if that’s saying no. An
+   initial response to open PRs should occur within [TBD] business days.
+1. A contribution should be completed through the efforts of the contributor and
+   maintainers and not be left abandoned. The maintainers are responsible see it
+   through to a conclusion.
+1. We should prioritize the things that people want based on feedback they
+   provide on issues. Users should know how to provide that feedback.
+1. Users should know exactly what to expect when interacting with the
+   maintainers. The next steps should be clearly defined.
+
+## Specification
+ 
+ ![Figure A: Flowchart of contribution guide review
+process.](contrib-guide-high-level.png "Figure A: Flowchart of contribution
+guide review process.")
+
+**Figure A: Flowchart of contribution guide review process.**
+
+We propose to make changes to the maintainer engagement and review process for
+feature requests and pull requests. The accompanying contributing guide must
+define what is required to move a feature request to be available for community
+contribution. It documents the review process and sets community expectations
+around communications to move a contribution forward. 
+
+Figure A demonstrates a high level view of the proposed process. It has three
+phases: intake, implementation, and review.
+
+### Intake
+
+The intake process identifies feature requests, through GitHub issues, where
+community contributions are welcome.
+
+The maintainers will actively review GitHub issues for feature requests to make
+this determination. The criteria used to make that decision include:
+
+1. The change will be made on part of the code base that the maintainers have
+   determined to be available. See the FAQ for more details.
```

Comment:
This list of unavailable parts of the codebase should be either in the specification or in an appendix (if the list is large). The FAQ is usually reserved for rationale on why a specific decision was made.

Reply:
done